### ISSUE TITLE
Connection Stats

### ISSUE THREAD

**Issue by murat-dogan** [paullouisageneau/libdatachannel#46]
I want to get some information about connection like in WebRTC Stats.
I need at least these values;

roundTripTime (like ping time)
bytesSent
bytesReceived
We can have bytesSent and bytesReceived values easily but I could not find a good way for calculating roundTripTime.

Do you have any idea, how to calculate roundTripTime?
Should we add stats function to libdatachannel?

**Comment by paullouisageneau** [comment]
`bytesSent` and `bytesReceived` are easy enough to compute, it only depends on the layer they should be computed for. My guess is it should be the bare data transferred on channels without headers and stuff.

The network RTT is computed by SCTP and seems to be exposed: https://tools.ietf.org/html/rfc6458#section-8.2.2
However, this won't be the ping time you get on a DataChannel under load due to the congestion control and send queue, more like the minimun you can expect for this connection. Is that what you mean?

**Comment by murat-dogan** [comment]
I think you are talking about
```
spinfo_srtt:  This contains the peer address's current smoothed
      round-trip time calculation in milliseconds.
```
I think stats should give an idea about connection speed so it could be enough.
Should we add this functionality to libdatachannel or leave it to user?

**Comment by paullouisageneau** [comment]
Yes, that's the one. Since in our case there is only one peer address for SCTP there is no ambiguity.

I think a getter for these on PeerConnection forwarding the call to SctpTransport would be good if you feel like adding it !

**Comment by murat-dogan** [comment]
Could you please check this commit ?
https://github.com/murat-dogan/libdatachannel/commit/94561ec7e50b796569ed919eb6f7fa3a52f255bb


I am not sure about bytesSent & bytesReceived.

**Comment by murat-dogan** [comment]
Resolved by pull request #50
